src/pvt.c

```diff
@@ -278,8 +279,240 @@ static u8 filter_solution(gnss_solution* soln, dops_t* dops)
+    assert(flag == 0);
+    if (removed_prn) {
+      *removed_prn = nav_meas[bad_sat].prn;
+    }
+    return 1;
+  } else {
+    return -1;
+  }
+}
+
+/* Calculate pvt solution, perform RAIM check, attempt to repair if needed.
+ *
+ * See calc_PVT for parameter meanings.
+ *
+ * Return values:
+ *    2: solution ok, but raim check was not used
+ *         (exactly 4 measurements, or explicitly disabled)
+ *
+ *    1: repaired solution, using one fewer observation
+ *       returns prn of removed measurement if removed_prn ptr is passed
+ *
+ *    0: inital solution ok
+ *
+ *   -1: repair failed
+ *   -2: not enough satellites to attempt repair
+ *   -3: pvt_iter didn't converge
+ *
+ *  Results stored in rx_state, H
+ */
+static s8 pvt_solve_raim(double rx_state[],
+                         const u8 n_used,
+                         const navigation_measurement_t nav_meas[n_used],
+                         bool disable_raim,
+                         double H[4][4],
+                         u8 *removed_prn,
+                         double residual)
+{
+  double omp[n_used];
+
+  assert(n_used <= MAX_CHANNELS);
+
+  const navigation_measurement_t *nav_meas_ptrs[n_used];
+  for (s8 i = 0; i < n_used; i++) {
+    nav_meas_ptrs[i] = &nav_meas[i];
+  }
+
+  s8 flag = pvt_iter(rx_state, n_used, nav_meas_ptrs, omp, H);
+
+  if (flag == -1) {
+    /* Iteration didn't converge. Don't attempt to repair; too CPU intensive. */
+    return -3;
+  }
+  if (flag >= 0 && (disable_raim || residual_test(n_used, omp, rx_state, &residual))) {
+    /* Solution ok, or raim check disabled. */
+    if (disable_raim || n_used == 4) {
+      /* Residual test couldn't have detected an error. */
+      return 2;
+    }
+    return 0;
+  } else {
+    if (n_used < 6) {
+      /* Not enough measurements to repair.
+       * 6 are needed because a 4 dimensional system is exactly constrained,
+       * so the bad measurement can't be detected.
+       */
+      return -2;
+    }
+    return pvt_repair(rx_state, n_used, nav_meas, omp, H, removed_prn);
+  }
+}
+
+/** Error strings for calc_PVT() negative (failure) return codes.
+ *  e.g. `pvt_err_msg[-ret - 1]`
+ *    where `ret` is the return value of calc_PVT(). */
+const char *pvt_err_msg[] = {
+  "PDOP too high",
+  "Altitude unreasonable",
+  "ITAR lockout",
+  "RAIM repair attempted, failed",
+  "RAIM repair impossible (not enough measurements)",
+  "Took too long to converge",
+  "Not enough measurements for solution (< 4)",
+};
+
+/** Try to calculate a single point gps solution
+ *
+ * \param n_used number of measurments
+ * \param nav_meas array of measurements
+ * \param disable_raim passing True will omit raim check/repair functionality
+ * \param soln output solution struct
+ * \param dops output doppler information
+ * \return Non-negative values indicate success; refer to pvt_solve_raim.
+ *         For negative values, refer to pvt_err_msg().
```

Comment:
`pvt_err_msg` isn't a function, I think you need to do `#pvt_err_msg` but not sure.